src/descriptor/sh.rs

```diff
@@ -205,52 +205,89 @@ impl<Pk: MiniscriptKey> Sh<Pk> {
+                4 * (varint_len(scriptsig_len) + scriptsig_len)
+            }
+            // add weighted script sig, len byte stays the same
+            ShInner::Wpkh(ref wpkh) => 4 * 23 + wpkh.max_satisfaction_weight(),
+            ShInner::Ms(ref ms) => {
+                let ss = ms.script_size();
+                let ps = push_opcode_size(ss);
+                let scriptsig_len = ps + ss + ms.max_satisfaction_size()?;
+                4 * (varint_len(scriptsig_len) + scriptsig_len)
+            }
+        })
+    }
 }
 
 impl<Pk: MiniscriptKey + ToPublicKey> Sh<Pk> {
-    /// Obtain the corresponding script pubkey for this descriptor
-    /// Non failing verion of [`DescriptorTrait::script_pubkey`] for this descriptor
-    pub fn spk(&self) -> Script {
+    /// Obtains the corresponding script pubkey for this descriptor.
+    pub fn script_pubkey(&self) -> Script {
         match self.inner {
-            ShInner::Wsh(ref wsh) => wsh.spk().to_p2sh(),
-            ShInner::Wpkh(ref wpkh) => wpkh.spk().to_p2sh(),
+            ShInner::Wsh(ref wsh) => wsh.script_pubkey().to_p2sh(),
+            ShInner::Wpkh(ref wpkh) => wpkh.script_pubkey().to_p2sh(),
             ShInner::SortedMulti(ref smv) => smv.encode().to_p2sh(),
             ShInner::Ms(ref ms) => ms.encode().to_p2sh(),
         }
     }
 
-    /// Obtain the corresponding script pubkey for this descriptor
-    /// Non failing verion of [`DescriptorTrait::address`] for this descriptor
-    pub fn addr(&self, network: bitcoin::Network) -> bitcoin::Address {
+    /// Obtains the corresponding address for this descriptor.
+    pub fn address(&self, network: Network) -> Address {
+        let addr = self.address_fallible(network);
+
+        // Size is checked in `check_global_consensus_validity`.
+        assert!(addr.is_ok());
+        addr.expect("only fails if size > MAX_SCRIPT_ELEMENT_SIZE")
+    }
+
+    fn address_fallible(&self, network: Network) -> Result<Address, Error> {
         match self.inner {
-            ShInner::Wsh(ref wsh) => {
-                bitcoin::Address::p2sh(&wsh.spk(), network).expect("Size checked in Miniscript")
-            }
-            ShInner::Wpkh(ref wpkh) => {
-                bitcoin::Address::p2sh(&wpkh.spk(), network).expect("Size checked in Miniscript")
-            }
-            ShInner::SortedMulti(ref smv) => {
-                bitcoin::Address::p2sh(&smv.encode(), network).expect("Size checked in Miniscript")
-            }
-            ShInner::Ms(ref ms) => {
-                bitcoin::Address::p2sh(&ms.encode(), network).expect("Size checked in Miniscript")
-            }
+            ShInner::Wsh(ref wsh) => Ok(Address::p2sh(&wsh.script_pubkey(), network)?),
+            ShInner::Wpkh(ref wpkh) => Ok(Address::p2sh(&wpkh.script_pubkey(), network)?),
+            ShInner::SortedMulti(ref smv) => Ok(Address::p2sh(&smv.encode(), network)?),
+            ShInner::Ms(ref ms) => Ok(Address::p2sh(&ms.encode(), network)?),
         }
```

Comment:
I think this would be easier to understand as:

```rust
        let script = match self.inner {
            ShInner::Wsh(ref wsh) => wsh.script_pubkey(),
            ShInner::Wpkh(ref wpkh) => wpkh.script_pubkey(),
            ShInner::SortedMulti(ref smv) => smv.encode(),
            ShInner::Ms(ref ms) => ms.encode(),
        };
	let address = Address::p2sh(&script, network)?;

	Ok(address)
```

Reply:
Nice, thanks.